### ISSUE TITLE
c-s with -log hdrfile does not work

### ISSUE THREAD

**Issue by asias** [scylladb/scylla-tools-java#308]
Scylla commit 934af9be52df9bae661a0d5c65c4e5411da992c9


```
$ cassandra-stress write no-warmup cl=ONE n=50000000 -schema 'replication(factor=2)' -port jmx=6868 -mode cql3 native -rate threads=200 throttle=20000/s -log hdrfile=c-s.hdr.txt  interval=10s -col 'size=FIXED(5) n=FIXED(8)' -pop seq=1500000000..250000000


===== Using optimized driver!!! =====                                                                                                                                                                                                                         
Connected to cluster: test, max pending requests per connection null, max connections per host 8
Datatacenter: dc1; Host: localhost/127.0.0.1; Rack: rack1
Datatacenter: dc1; Host: /127.0.0.3; Rack: rack1
WARN  09:17:28,084 Not using advanced port-based shard awareness with /127.0.0.3:9042 because we're missing port-based shard awareness port on the server
WARN  09:17:28,136 Not using advanced port-based shard awareness with localhost/127.0.0.1:9042 because we're missing port-based shard awareness port on the server
Created keyspaces. Sleeping 1s for propagation.
Sleeping 2s...
Running WRITE with 200 threads for 50000000 iteration
Failed to connect over JMX; not collecting these stats
type       total ops,    op/s,    pk/s,   row/s,    mean,     med,     .95,     .99,    .999,     max,   time,   stderr, errors,  gc: #,  max ms,  sum ms,  sdv ms,      mb
Exception in thread "StressMetrics" java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
        at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:135)
        at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:154)
        at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:175)
        at org.apache.cassandra.stress.report.StressMetrics.logHistogram(StressMetrics.java:336)
        at org.apache.cassandra.stress.report.StressMetrics.logHistograms(StressMetrics.java:320)
        at org.apache.cassandra.stress.report.StressMetrics.recordInterval(StressMetrics.java:253)
        at org.apache.cassandra.stress.report.StressMetrics.reportingLoop(StressMetrics.java:172)
        at org.apache.cassandra.stress.report.StressMetrics.lambda$new$1(StressMetrics.java:124)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 9 more

```

**Comment by asias** [comment]
I started latest Apache Cassandra 4.0.6  server and run c-s tool from 4.0.6.
https://www.apache.org/dyn/closer.lua/cassandra/4.0.6/apache-cassandra-4.0.6-bin.tar.gz


I saw similar error

```
Exception in thread "StressMetrics" java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
        at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:135)
        at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:154)
        at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:175)
        at org.apache.cassandra.stress.report.StressMetrics.logHistogram(StressMetrics.java:341)
        at org.apache.cassandra.stress.report.StressMetrics.logHistograms(StressMetrics.java:325)
        at org.apache.cassandra.stress.report.StressMetrics.recordInterval(StressMetrics.java:258)
        at org.apache.cassandra.stress.report.StressMetrics.reportingLoop(StressMetrics.java:177)
        at org.apache.cassandra.stress.report.StressMetrics.lambda$new$1(StressMetrics.java:130)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 9 more
```

**Comment by asias** [comment]
Good news. I managed to get it work by:

Switching back to the old 1.8.0 jdk on fedora 35.

sudo ln -sf /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.345.b01-1.fc35.x86_64/jre/bin/java /etc/alternatives/java


I am using the c-s from scylla version 6799e766ca9594b1d793a5aa1a781dc63e78e734
cassandra-stress write no-warmup cl=ONE n=50000000 -schema 'replication(factor=2)' -mode cql3 native -rate threads=200 throttle=20000/s -log hdrfile=hdr.txt interval=10s

**Comment by michoecho** [comment]
c-s respects `JAVA_HOME`, so you can just run `JAVA_HOME=/usr/lib/jvm/java-1.8.0 cassandra-stress`. That's how I have always been working around this dependency.

**Comment by avikivity** [comment]
> I started latest Apache Cassandra 4.0.6 server and run c-s tool from 4.0.6. https://www.apache.org/dyn/closer.lua/cassandra/4.0.6/apache-cassandra-4.0.6-bin.tar.gz
> 
> I saw similar error
> 
> ```
> Exception in thread "StressMetrics" java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
>         at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:135)
>         at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:154)
>         at org.HdrHistogram.HistogramLogWriter.outputIntervalHistogram(HistogramLogWriter.java:175)
>         at org.apache.cassandra.stress.report.StressMetrics.logHistogram(StressMetrics.java:341)
>         at org.apache.cassandra.stress.report.StressMetrics.logHistograms(StressMetrics.java:325)
>         at org.apache.cassandra.stress.report.StressMetrics.recordInterval(StressMetrics.java:258)
>         at org.apache.cassandra.stress.report.StressMetrics.reportingLoop(StressMetrics.java:177)
>         at org.apache.cassandra.stress.report.StressMetrics.lambda$new$1(StressMetrics.java:130)
>         at java.base/java.lang.Thread.run(Thread.java:829)
> Caused by: java.lang.ClassNotFoundException: javax.xml.bind.DatatypeConverter
>         at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
>         at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
>         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
>         ... 9 more
> ```

Please report it upstream.

**Comment by michoecho** [comment]
Posted a fix to #312. Not sure how it works, though.
